src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -127,10 +127,24 @@ fun inferExpressionType(expr: RsExpr): RustType {
                 RustUnknownType
         }
 
+        is RsArrayExpr -> inferArrayType(expr)
+
         else -> RustUnknownType
     }
 }
 
+private fun inferArrayType(expr: RsArrayExpr): RustType {
+    val (elementType, size) = if (expr.semicolon != null) {
+        if (expr.exprList.size != 2) return RustUnknownType
```

Comment:
>Maybe drop it at all?

Yeah, it should be safe to just `check(expr.exprList.size == 2)`. Better still would be to add general accessors to `RsArrayExpr` like `RsArrayExpr.initializer: RsExpr?`, `RsArrayExpr.size: RsExpr?` and ``RsArrayExpr.arrayElements: List<RsExpr>?`.